webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/ncf/EditorTechniqueReader.scala

```diff
@@ -251,13 +254,14 @@ final case class JsonParameterConstraint(
     min_length.map(MinLength.apply).toList :::
     regex.map(MatchRegex.apply).toList :::
     not_regex.map(NotMatchRegex.apply).toList :::
-    select.map(FromList.apply).toList
+    select.map(_.map(_.value)).map(FromList.apply).toList
   }
 }
 
 object GenericMethodSerialization {
 
   implicit val decodeJsonParameterConstraint:          JsonDecoder[JsonParameterConstraint]          = DeriveJsonDecoder.gen
+  implicit val decodeJsonParameterConstraintSelect:    JsonDecoder[JsonParameterConstraintSelect]    = DeriveJsonDecoder.gen
```

Comment:
Should be defined before `JsonParameterConstraint` since it's used as a field in it (or put `implicit lazy val` to get at-nead initialization)

Reply:
thanks!